core/src/main/scala/org/apache/spark/executor/TaskMetrics.scala

```diff
@@ -240,10 +284,18 @@ class ShuffleWriteMetrics extends Serializable {
   /**
    * Number of bytes written for the shuffle by this task
    */
-  @volatile var shuffleBytesWritten: Long = _
-
+  @volatile private var _shuffleBytesWritten: Long = _
+  def shuffleBytesWritten = _shuffleBytesWritten
+  private[spark] def incShuffleBytesWritten(value: Long) = _shuffleBytesWritten += value
+  private[spark] def decShuffleBytesWritten(value: Long) = _shuffleBytesWritten -= value
+  
   /**
    * Time the task spent blocking on writes to disk or buffer cache, in nanoseconds
    */
-  @volatile var shuffleWriteTime: Long = _
```

Comment:
`_shuffleWriteTime` can only get incremented from a single thread.  It's marked volatile so that other threads can read from it. Using an `AtomicLong` would unnecessarily bloat the size of the task results.  I probably should have documented this access pattern better.